typeid/typeid-go/constructors.go

```diff
@@ -98,8 +98,31 @@ func split(id string) (string, string, error) {
 	}
 }
 
-// FromUUID encodes the given UUID (in hex string form) as a TypeID with the given prefix.
-func FromUUID[T Subtype, PT SubtypePtr[T]](prefix string, uidStr string) (T, error) {
+// FromUUID encodes the given UUID (in hex string form) as a TypeID
+func FromUUID[T Subtype, PT SubtypePtr[T]](uidStr string) (T, error) {
+	return fromUUID[T, PT](defaultPrefix[T](), uidStr)
```

Comment:
Could you add:
```
	if isAnyID[T]() {
		var id T
		return id, errors.New("constructor error: use FromUUIDWithPrefix(), FromUUID() is for Subtypes")
	}
```

Is just to avoid the case of someone calling `FromUUID[AnyID](...)`, that results in a typeid with no prefix, which is probably unintended.

Same in FromUUIDBytes()